routes/challenge.js

```diff
@@ -190,8 +190,12 @@ exports.edit = function(req, res) {
         'end':         new Date(req.body.end.replace(pattern, '$3-$2-$1'))
     }};
     Challenges.update(conditions, update, function (err, num) {
-      console.log("* Owner made changes to challenge " + req.body.name);
-      res.redirect('/challenges/' + req.body.name.replace(/\s+/g, '') + '/admin');
+      if (err) {
+        console.log("Failed to edit challenge name to: " + req.body.name)
```

Comment:
We can use https://github.com/jonnyreeves/js-logger#log-handler-functions -> and add authentication to the route :)